go/extractor/extractor.go

```diff
@@ -174,16 +177,17 @@ func ExtractWithFlags(buildFlags []string, patterns []string) error {
 		return true
 	}, func(pkg *packages.Package) {
 		for root, _ := range wantedRoots {
-			relDir, err := filepath.Rel(root, pkgDirs[pkg.PkgPath])
+			pkgInfo := pkgInfos[pkg.PkgPath]
+			relDir, err := filepath.Rel(root, pkgInfo.PkgDir)
 			if err != nil || noExtractRe.MatchString(relDir) {
 				// if the path can't be made relative or matches the noExtract regexp skip it
 				continue
 			}
 
 			extraction.extractPackage(pkg)
 
-			if pkgRoots[pkg.PkgPath] != "" {
-				modPath := filepath.Join(pkgRoots[pkg.PkgPath], "go.mod")
+			if pkgInfo.ModDir != "" {
```

Comment:
Because this doesn't fall back to `PkgDir` as before, we don't extract `go.mod` files in subdirectories when we couldn't work out their `ModDir`. Would it be simplest to just populate `ModDir` with `PkgDir` as the old code did, to avoid needlessly confusing users? It's fair to say if there's a `go.mod` and we couldn't work out the `ModDir` we're probably building the project wrongly anyhow, but we may as well limit confusion on top of that.

Reply:
Yeah, this was an oversight on my part. Updated.